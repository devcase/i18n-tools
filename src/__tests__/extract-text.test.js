import extractText from '../extract-text'

test("test 1", () => {
    const code = `
        var x = "Este texto precisa ser extraído"
    `
    const results = extractText(code);
    expect(results).toHaveLength(1)

    expect(results[0]).toMatchObject({
        key: "este-texto-precisa-ser-extraido",
        value: "Este texto precisa ser extraído",
        hash: 2951570100
    })
})

test("test com jsx", () => {
    const code = `
        import React from 'react'
        
        function Component(props) {
            return <div 
                label="Este é um label e precisa ser extraído" 
                className="mb-2 ignorethis"
                max="3">
                Este texto precisa ser extraído
            </div>;
        }
    `
    const results = extractText(code)
    expect(results).toMatchSnapshot()

})