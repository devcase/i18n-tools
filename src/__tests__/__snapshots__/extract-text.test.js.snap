// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test com jsx 1`] = `
Array [
  Object {
    "hash": 4261676902,
    "key": "este-e-um-label-e-precisa-ser-extraido",
    "value": "Este é um label e precisa ser extraído",
  },
  Object {
    "hash": 2951570100,
    "key": "este-texto-precisa-ser-extraido",
    "value": "Este texto precisa ser extraído",
  },
]
`;
